crates/sui-types/src/eth_dwallet.rs

```diff
@@ -0,0 +1,44 @@
+use crate::base_types::ObjectID;
+use anyhow::anyhow;
+use move_core_types::ident_str;
+use move_core_types::identifier::IdentStr;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::id::{ID, UID};
+
+/// Constants that define the names of the Move modules and functions used in the Ethereum
+/// messages verification process.
+pub const ETH_DWALLET_MODULE_NAME: &IdentStr = ident_str!("eth_dwallet");
+pub const ETHEREUM_STATE_MODULE_NAME: &IdentStr = ident_str!("ethereum_state");
+pub const LATEST_ETH_STATE_STRUCT_NAME: &IdentStr = ident_str!("LatestEthereumState");
+pub const ETH_STATE_STRUCT_NAME: &IdentStr = ident_str!("EthState");
+pub const INIT_STATE_FUNC_NAME: &IdentStr = ident_str!("init_state");
+pub const CREATE_ETH_DWALLET_CAP_FUNC_NAME: &IdentStr = ident_str!("create_eth_dwallet_cap");
+pub const VERIFY_ETH_STATE_FUNC_NAME: &IdentStr = ident_str!("verify_new_state");
+pub const APPROVE_MESSAGE_FUNC_NAME: &IdentStr = ident_str!("approve_message");
+
+/// Rust version of the Move [`dwallet_system::eth_dwallet::EthDWalletCap`] type.
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, Eq, PartialEq)]
+pub struct EthDWalletCap {
```

Comment:
```suggestion
pub struct EthereumDWalletCap {
```
Please use fully abbreviated names